### PR TITLE
fix(lambda-alarms): Add alarms support to PocketEventBridgeWithLambdaTarget module

### PR DESCRIPTION
**Pull request by kkyeboah**
## Goal

- Add alarms support to PocketEventBridgeWithLambdaTarget module
- Stop using escape hatch for dimensions in other modules (fixed in `cdktf` 0.0.19)

### CODE REVIEW

File: src/base/ApplicationAutoscaling.ts [hunk 1]
```diff
@@ -162,24 +162,15 @@ export class ApplicationAutoscaling extends Resource {
...
-      dimensions: {},
+      dimensions: {
+        ClusterName: config.ecsClusterName,
+        ServiceName: config.ecsServiceName,
+      },
       alarmActions: [applicationScaleOut.arn],
       tags: config.tags,
     });
 
-    // This override is related to CDK bug: https://github.com/hashicorp/terraform-cdk/issues/235
```
Comment (kkyeboah) [inline src/base/ApplicationAutoscaling.ts:170]: Fixed in `cdktf` version 0.0.19

File: src/base/ApplicationAutoscaling.ts [hunk 2]
```diff
@@ -190,15 +181,12 @@ export class ApplicationAutoscaling extends Resource {
...
-      dimensions: {},
+      dimensions: {
+        ClusterName: config.ecsClusterName,
+        ServiceName: config.ecsServiceName,
+      },
       alarmActions: [applicationScaleIn.arn],
       tags: config.tags,
     });
-
-    // This override is related to CDK bug: https://github.com/hashicorp/terraform-cdk/issues/235
```
Comment (kkyeboah) [inline src/base/ApplicationAutoscaling.ts:198]: Fixed in `cdktf` version 0.0.19

File: src/pocket/PocketALBApplication.ts [hunk 3]
```diff
@@ -839,46 +829,9 @@ export class PocketALBApplication extends Resource {
 
   private createAlarms(alarms: CloudwatchMetricAlarmConfig[]): void {
     alarms.forEach((alarmConfig) => {
-      // Deep copy the alarm configurations to a new object to enable deletion of properties since they are
```
Comment (kkyeboah) [inline src/pocket/PocketALBApplication.ts:842]: Fixed in cdktf version 0.0.19

File: src/pocket/PocketEventBridgeWithLambdaTarget.ts [hunk 4]
```diff
@@ -6,75 +6,228 @@ import {
   LAMBDA_RUNTIMES,
```
Comment (kkyeboah) [inline src/pocket/PocketEventBridgeWithLambdaTarget.ts:6]: This class is the focus of the PR

**Review by jpetto — APPROVED** [review]

**Pull request merged**